src/lib/raft/raft.h

```diff
@@ -191,6 +198,15 @@ struct raft {
 	 * Usually even in one function.
 	 */
 	bool is_broadcast_scheduled;
+	/**
+	 * Whether this instance has a quorum of connections to and from the
+	 * remote peers or not.
+	 */
+	bool have_quorum;
```

Comment:
You never actually use this flag in raft.c as something related to quorum. It looks like just another flag to enable/disable candidate. Is it possible that you could drop it entirely and instead call `raft_cfg_is_candidate(false)` when quorum is lost and `raft_cfg_is_candidate(true)` when the quorum is present?
Then you would need to change stuff in box - add a new function `box_raft_update_candidate()`. It would check the quorum flag in box and `box_election_mode` global variable.

Although then you would need to keep the leader watcher timer running even for non-candidates to track when the leader doesn't send anything for too long, right?

Reply:
Done, please see v2. Basically everything in `src/lib/raft.c` and `box_raft_cfg_election_mode()`, `box_raft_notify_have_quorum()` in `src/box/raft.c`